web/client-api/src/main/resources/io/deephaven/web/super/io/deephaven/engine/rowset/WebRowSetImpl.java

```diff
@@ -0,0 +1,89 @@
+package io.deephaven.engine.rowset;
+
+import io.deephaven.util.datastructures.LongAbortableConsumer;
+import io.deephaven.util.datastructures.LongRangeConsumer;
+import io.deephaven.web.shared.data.RangeSet;
+
+import java.util.PrimitiveIterator;
+
+final class WebRowSetImpl implements RowSet, WritableRowSet {
+    private final RangeSet rangeSet;
+
+    WebRowSetImpl(RangeSet rangeSet) {
+        this.rangeSet = rangeSet;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return rangeSet.isEmpty();
+    }
+
+    @Override
+    public long lastRowKey() {
+        return rangeSet.getLastRow();
+    }
+
+    @Override
+    public boolean forEachRowKey(LongAbortableConsumer lac) {
+        PrimitiveIterator.OfLong iter = rangeSet.indexIterator();
+        while (iter.hasNext()) {
+            long key = iter.nextLong();
+            if (!lac.accept(key)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public void forAllRowKeyRanges(LongRangeConsumer lrc) {
+        rangeSet.rangeIterator().forEachRemaining(r -> {
+            lrc.accept(r.getFirst(), r.getLast());
+        });
+    }
+
+    @Override
```

Comment:
Yes, I'll make this change.

At this time however, the entire RowSet emulation is unused, and only exists so that the BSGI types can compile at all, since we reference some static members of those types. These will be needed in the future when the JS API uses DoPut (or publish via DoExchange...) uses the shared JVM code.